### ISSUE TITLE
Data Model orchestration

### ISSUE THREAD

**Issue by beasteers** [directus-community/helm-chart#10]
I'm trying to deploy directus as part of an automatic CI/CD pipeline (with terraform) so that instances can be brought up and down automatically.

Ideally, I'd like to synchronize directus's snapshot so that it can be instantiated from a github repo. I'm thinking of adding some sidecars such as [git-sync](https://github.com/kubernetes/git-sync) and another sidecar that runs `npx directus schema snapshot --yes ./snapshot.yaml` periodically to dump the current schema so that it can be picked up by git-sync and uploaded to github.

So all that needs to be done to the chart for that part is to add the ability to add sidecars to the deployment.

----

But the second part is how to apply the snapshot to directus. The obvious answer is to use an initcontainer and to mount the snapshot as a config map, but it would need to be careful to not overwrite changes that a user makes through the UI.

That leads me to think that there is room for an extension that synchronizes the schema with a local file
 - Listen to data model change events -> write snapshot to file
     - (I don't see data model changes in the events listed 😥)
 - Watch for file change events -> apply snapshot
     - (Can use `kiwigrid/k8s-sidecar` with [REQ_URL](https://github.com/kiwigrid/k8s-sidecar#:~:text=integer-,REQ_URL,-URL%20to%20which) and the extension could have a custom endpoint to do the apply)

Then you'd use git-sync to sync the file with github.

I'm wondering if anyone else has a similar use case?

tl;dr - looking for a robust way to sync the directus data model with git

**Comment by kyuumeitai** [comment]
I'm a total k8s noob but I was wondering the same thing.

**Comment by kyuumeitai** [comment]
Hi @beasteers ! is this relevant to what you're trying to achieve?

https://github.com/directus/directus/pull/14713

**Comment by beasteers** [comment]
Oooo that's great! I love when features you've been searching for were merged "last week" 🌈 Thanks so much for the heads up 🙏🏻

There's still a question of synchronization, but the API endpoints make it much easier to decouple that functionality from the container.

**Comment by fr3fou** [comment]
Have you figured anything new out yet? @beasteers 
I was thinking of a similar idea, but instead of using sidecar for applying the schema snapshots, I'd use ArgoCD's `PreSync` resource hook `https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks` to run a container which applies the snapshot using the Directus API.

I haven't figured out a way to reliably dump the snapshots though -- the schema can be changed both in a remote deployment (e.g staging / dev) or locally.
